Collections

ali01 opened the pull request.

ali01 committed:
basic: DirectoryDatastore => DirectoryTreeDatastore

ali01 committed:
basic: added DirectoryDatastore + tests

jbenet commented:
@ali01 is this good to go?

jbenet commented:
@ali01 bump

jbenet commented:
Seems like it. tests pass.

Warning, this breaks backward compatibility: `DirectoryDatastore` has been renamed.

jbenet committed:
Fixed some semantic issues

- initializing directories does not wipe out old dirs.
- only append entries if not already there
- generator yields Key instances
- removing entry removes all occurrences

@ali01

jbenet commented:
Closes #6
